### PR TITLE
Address #166: Set history file max size from inputrc via JLine

### PR DESCRIPTION
**Pull request by timmc-bcov**
Set history file max size based on inputrc, using newer jline2 to support this configuration.

Please let me know if there are any tests you would like to see.

### CODE REVIEW

**Comment by timmc-bcov** [comment]
^ I can't seem to get any detail from Travis about that failing build.

**Comment by timmc-bcov** [comment]
OK, this is failing because there's no `.inputrc`, which I find surprising but I guess not all that surprising. Might be a jline2 bug. -.-

**Comment by trptcolin** [comment]
I have an `inputrc` locally and was able to reproduce the failures - I see via `lein trampoline run` that there's a stack trace on startup (unfortunately it's missing from these test outputs because it's not asserting against `@fake-err` being empty - that might be a good addition to all these integration tests). Here's what I get:

```
NumberFormatException null
        java.lang.Integer.parseInt (Integer.java:542)
        java.lang.Integer.parseInt (Integer.java:615)
        reply.reader.simple-jline/default-history-size (simple_jline.clj:76)
        reply.reader.simple-jline/setup-console-reader (simple_jline.clj:98)
        reply.reader.simple-jline/get-input-line (simple_jline.clj:131)
        clojure.lang.Atom.swap (Atom.java:37)
        clojure.core/swap! (core.clj:2232)
        reply.reader.simple-jline/safe-read-line (simple_jline.clj:170)
        reply.reader.simple-jline/safe-read-line (simple_jline.clj:177)
        clojure.core/apply (core.clj:626)
        clojure.core/partial/fn--4228 (core.clj:2468)
        reply.eval-modes.nrepl/run-repl (nrepl.clj:133)
```

So I think it just needs a failsafe where if `"history-size"` isn't present, it doesn't try to do the NPE-causing `Integer/parseInt` - maybe `some->` instead of `->`?

**Comment by timmc-bcov** [comment]
Ah, thanks! That explains why I was having trouble isolating this error. I'll put in a fix.

File: spec/reply/integration_spec.clj [hunk 1]
```diff
@@ -67,27 +68,31 @@
...
                              :output-stream @fake-out})
+    (should= "" (str @fake-err))
     (should-contain #"Unmatched delimiter.+\n.+\n.+Unmatched delimiter" (str @fake-out)))
 
   (it "does not print an error when given empty input lines"
     (main/launch-standalone {:input-stream
                              (java.io.ByteArrayInputStream.
                                (.getBytes "\n\n\n"))
                              :output-stream @fake-out})
+    (should= "" (str @fake-err))
```
Comment (timmc-bcov) [inline spec/reply/integration_spec.clj]: I wonder if this addition actually obviates these existing `should-not-contain` checks on stdout, or if those errors might really appear on stdout if a bug is present.

**Comment by timmc-bcov** [comment]
OK, this works better, and I fixed an additional issue around possibly malformed `history-size` values (non-numbers) in .inputrc files.

File: spec/reply/integration_spec.clj (hunk 1 above)
Comment (trptcolin) [reply spec/reply/integration_spec.clj]: Hrm, great question - I suspect you're right that these should've been checks against `fake-err`. In any case, non-blocker. Thanks!

**Comment by trptcolin** [comment]
Thanks!! 🎉

**Comment by timmc-bcov** [comment]
And thank you!
